pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -1371,6 +1372,11 @@ func (l *LibvirtDomainManager) SyncVMI(vmi *v1.VirtualMachineInstance, useEmulat
 	// To make sure, that we set the right qemu wrapper arguments,
 	// we update the domain XML whenever a VirtualMachineInstance was already defined but not running
 	if !newDomain && cli.IsDown(domState) {
+		domain, err = l.preStartHook(vmi, domain)
```

Comment:
I think you need to invoke just `network.SetupNetworkInterfacesPhase2(vmi, domain)`, not the whole `preStartHook`.

Reply:
But the problem in general is that on retries we don't update the dom xml from libvirt we re-create it again right ? so we have to put back the bits that are set on preStartHook.